vllm/v1/engine/core.py

```diff
@@ -107,15 +115,17 @@ def add_request(self, request: EngineCoreRequest):
                 request.mm_inputs, request.mm_hashes)
 
         req = Request.from_engine_core_request(request)
+        if req.use_guided_decoding:
+            self.use_guided_decoding = True
+            # Start grammar compilation asynchronously
+            self.guided_decoding_manager.should_cache(req)
+        else:
+            self.use_guided_decoding = False
```

Comment:
If we have a request come in that uses guided decoding but then another request right after that doesn't, this disables guided decoding for the whole EngineCore? I think there isn't a need for EngineCore to have `use_guided_decoding`, this can be managed within `guided_decoding_manager`

Reply:
sg. I have reflected this change